lib/yarv/send.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module YARV
+  # ### Summary
+  #
+  # `send` invokes a method with a block.
+  #
+  # ### TracePoint
+  #
+  # `send` does not dispatch any events.
+  #
+  # ### Usage
+  #
+  # ~~~ruby
+  # "hello".tap { |i| p i }
+  #
+  # # == disasm: #<ISeq:<main>@-e:1 (1,0)-(1,23)> (catch: FALSE)
+  # # 0000 putstring                              "hello"                   (   1)[Li]
+  # # 0002 send                                   <calldata!mid:tap, argc:0>, block in <main>
+  # # 0005 leave
+  # #
+  # # == disasm: #<ISeq:block in <main>@-e:1 (1,12)-(1,23)> (catch: FALSE)
+  # # local table (size: 1, argc: 1 [opts: 0, rest: -1, post: 0, block: -1, kw: -1@-1, kwrest: -1])
+  # # [ 1] i@0<Arg>
+  # # 0000 putself                                                          (   1)[LiBc]
+  # # 0001 getlocal_WC_0                          i@0
+  # # 0003 opt_send_without_block                 <calldata!mid:p, argc:1, FCALL|ARGS_SIMPLE>
+  # # 0005 leave                                  [Br]
+  # # ~~~
+  #
+  class Send
+    attr_reader :call_data, :block_iseq
+
+    def initialize(call_data, block_iseq)
```

Comment:
Is there a case where `Send` doesn't receive a block_iseq? Or is it always there? Maybe try turning off some of the compilation options.

Reply:
With `specialized_instruction: false`, I do get a `nil` block_iseq (e.g. `[:send, {:mid=>:p, :flag=>20, :orig_argc=>1}, nil]`). 

Am thinking whether to pass a `nil` here, or to have something like a `NullInstructionSequence`, though I think I'll have a better answer when I fix the logic https://github.com/kddnewton/yarv/pull/134/files#r870906290